This projects aims to analyze the sales data from a German eBay website.

In [1]:
import pandas as pd
import numpy as np
autos= pd.read_csv('autos.csv', encoding = 'Latin-1')

In [2]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType    price   abtest vehicleType  yearOfRegistration  \
0      privat   Angebot   $5,000  control         bus                2004   
1      privat   Angebot   $8,500  control   limousine                1997   
2      privat   Angebot   $8,990     test   limousine                2009   
3      privat   Angebot   $4,350  control  kleinwagen                2007   
4      privat   Angebot   $1,350     test       kombi                2003   
...       ...       ...      ...      ...         ...                 ...   
49995  privat   Angebot  $24,900  control   limousine                2011   
49996  privat   Angebot   $1,980  control      cabrio                1996   
49997  privat   Angebot  $13,200     test      cabrio                2014   
49998  privat   Angebot  $22,900  control       kombi                2013   
49999  privat   Angebot   $1,250  control   limousine                1996   

         gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0        manuell      158  andere  150,000km                    3      lpg   
1      automatik      286     7er  150,000km                    6   benzin   
2        manuell      102    golf   70,000km                    7   benzin   
3      automatik       71  fortwo   70,000km                    6   benzin   
4        manuell        0   focus  150,000km                    7   benzin   
...          ...      ...     ...        ...                  ...      ...   
49995  automatik      239      q5  100,000km                    1   diesel   
49996    manuell       75   astra  150,000km                    5   benzin   
49997  automatik       69     500    5,000km                   11   benzin   
49998    manuell      150      a3   40,000km                   11   diesel   
49999    manuell      101  vectra  150,000km                    1   benzin   

            brand notRepairedDamage          dateCreated  nrOfPictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postalCode             lastSeen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
4999

In [3]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

In [4]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

Convert the column names from camelcase to snakecase and reword some of the column names based on the data dictionary to be more descriptive.

In [5]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [6]:
autos.columns = ['date_crawled', 'name', 'seller', 'offertype', 'price', 'abtest',
       'vehicleType', 'registration_year', 'gearbox', 'powerPS', 'model',
       'odometer', 'registration_month', 'fuelType', 'brand',
       'unrepaired_damage', 'ad_created', 'nrofpictures', 'postalCode',
       'last_seen']

In [7]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offertype   price   abtest vehicleType  registration_year  \
0  privat   Angebot  $5,000  control         bus               2004   
1  privat   Angebot  $8,500  control   limousine               1997   
2  privat   Angebot  $8,990     test   limousine               2009   
3  privat   Angebot  $4,350  control  kleinwagen               2007   
4  privat   Angebot  $1,350     test       kombi               2003   

     gearbox  powerPS   model   odometer  registration_month fuelType  \
0    manuell      158  andere  150,000km                   3      lpg   
1  automatik      286     7er  150,000km                   6   benzin   
2    manuell      102    golf   70,000km                   7   benzin   
3  automatik       71  fortwo   70,000km                   6   benzin   
4    manuell        0   focus  150,000km                   7   benzin   

        brand unrepaired_damage           ad_created  nrofpictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode            last_seen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

Checking for other data cleaning work that need to be done:
1. columns that contain the same values;
2. numeric data stored as text

In [8]:
autos.describe(include='all')

date_crawled         name  seller offertype  price abtest  \
count                 50000        50000   50000     50000  50000  50000   
unique                48213        38754       2         2   2357      2   
top     2016-03-23 18:39:34  Ford_Fiesta  privat   Angebot     $0   test   
freq                      3           78   49999     49999   1421  25756   
mean                    NaN          NaN     NaN       NaN    NaN    NaN   
std                     NaN          NaN     NaN       NaN    NaN    NaN   
min                     NaN          NaN     NaN       NaN    NaN    NaN   
25%                     NaN          NaN     NaN       NaN    NaN    NaN   
50%                     NaN          NaN     NaN       NaN    NaN    NaN   
75%                     NaN          NaN     NaN       NaN    NaN    NaN   
max                     NaN          NaN     NaN       NaN    NaN    NaN   

       vehicleType  registration_year  gearbox       powerPS  model  \
count        44905       50000.000000    47320  50000.000000  47242   
unique           8                NaN        2           NaN    245   
top      limousine                NaN  manuell           NaN   golf   
freq         12859                NaN    36993           NaN   4024   
mean           NaN        2005.073280      NaN    116.355920    NaN   
std            NaN         105.712813      NaN    209.216627    NaN   
min            NaN        1000.000000      NaN      0.000000    NaN   
25%            NaN        1999.000000      NaN     70.000000    NaN   
50%            NaN        2003.000000      NaN    105.000000    NaN   
75%            NaN        2008.000000      NaN    150.000000    NaN   
max            NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuelType       brand unrepaired_damage  \
count       50000        50000.000000    45518       50000             40171   
unique         13                 NaN        7          40                 2   
top     150,000km                 NaN   benzin  volkswagen              nein   
freq        32424                 NaN    30107       10687             35232   
mean          NaN            5.723360      NaN         NaN               NaN   
std           NaN            3.711984      NaN         NaN               NaN   
min           NaN            0.000000      NaN         NaN               NaN   
25%           NaN            3.000000      NaN         NaN               NaN   
50%           NaN            6.000000      NaN         NaN               NaN   
75%           NaN            9.000000      NaN         NaN               NaN   
max           NaN           12.000000      NaN         NaN               NaN   

                 ad_created  nrofpictures    postalCode            last_seen  
count                 50000       50000.0  50000.000000                50000  
unique                   76           NaN           NaN                39481  
top     2016-04-03 00:00:00           NaN           NaN  2016-04-07 06:17:27  
freq                   1946           NaN           NaN                    8  
mean                    NaN           0.0  50813.627300                  NaN  
std                     NaN           0.0  25779.747957                  NaN  
min                     NaN           0.0   1067.000000                  NaN  
25%                     NaN           0.0  30451.000000                  NaN  
50%                     NaN           0.0  49577.000000                  NaN  
75%                     NaN           0.0  71540.000000                  NaN  
max                     NaN           0.0  99998.000000                  NaN

From the table above, we know that:
1. 'seller', 'offer_type', 'nr_of_pictures' have mostly one value that are candidates to be dropped
2.  'year_of_registration', 'powerPS', month_of_registrationthat need more investigation
3. 'price', 'odometer' has numeric data stored as text that needs to be cleaned

In [9]:
# convert price column to numeric datatype

autos["price"] = autos["price"].str.replace("$","").str.replace(",","")
autos["price"]= autos["price"].astype(int)

In [10]:
autos["price"].unique()

array([ 5000,  8500,  8990, ...,   385, 22200, 16995])

In [11]:
#convert odometer to numeric datatype

autos["odometer"] = autos["odometer"].str.replace("km","").str.replace(",","")
autos["odometer"] = autos["odometer"].astype(int)

In [12]:
autos["odometer"].unique()

array([150000,  70000,  50000,  80000,  10000,  30000, 125000,  90000,
        20000,  60000,   5000, 100000,  40000])

In [13]:
autos.rename({'odometer':"odometer_km"},axis = 1, inplace =True)

In [14]:
autos["odometer_km"].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [15]:
autos["odometer_km"].value_counts().sort_index(ascending=True)

5000        967
10000       264
20000       784
30000       789
40000       819
50000      1027
60000      1164
70000      1230
80000      1436
90000      1757
100000     2169
125000     5170
150000    32424
Name: odometer_km, dtype: int64

In [16]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [17]:
autos["price"].value_counts().sort_index(ascending=False).head(15)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
Name: price, dtype: int64

In [18]:
autos["price"].value_counts().sort_index(ascending=False).tail(15)

18       1
17       3
15       2
14       1
13       2
12       3
11       2
10       7
9        1
8        1
5        2
3        1
2        3
1      156
0     1421
Name: price, dtype: int64

In [19]:
autos = autos[autos["price"].between(1000,1000000)]
autos

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offertype  price   abtest vehicleType  registration_year  \
0      privat   Angebot   5000  control         bus               2004   
1      privat   Angebot   8500  control   limousine               1997   
2      privat   Angebot   8990     test   limousine               2009   
3      privat   Angebot   4350  control  kleinwagen               2007   
4      privat   Angebot   1350     test       kombi               2003   
...       ...       ...    ...      ...         ...                ...   
49995  privat   Angebot  24900  control   limousine               2011   
49996  privat   Angebot   1980  control      cabrio               1996   
49997  privat   Angebot  13200     test      cabrio               2014   
49998  privat   Angebot  22900  control       kombi               2013   
49999  privat   Angebot   1250  control   limousine               1996   

         gearbox  powerPS   model  odometer_km  registration_month fuelType  \
0        manuell      158  andere       150000                   3      lpg   
1      automatik      286     7er       150000                   6   benzin   
2        manuell      102    golf        70000                   7   benzin   
3      automatik       71  fortwo        70000                   6   benzin   
4        manuell        0   focus       150000                   7   benzin   
...          ...      ...     ...          ...                 ...      ...   
49995  automatik      239      q5       100000                   1   diesel   
49996    manuell       75   astra       150000                   5   benzin   
49997  automatik       69     500         5000                  11   benzin   
49998    manuell      150      a3        40000                  11   diesel   
49999    manuell      101  vectra       150000                   1   benzin   

            brand unrepaired_damage           ad_created  nrofpictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postalCode            last_seen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
49995       82131  2016-04-

Note: It's unrealistics to have price blanks as 0, namely having cars to be sold for free. Thus I choose to set the price filter from 1k to 10k.

In [20]:
autos[['date_crawled','ad_created','last_seen'][0:5]]

date_crawled           ad_created            last_seen
0      2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1      2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2      2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3      2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4      2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50
...                    ...                  ...                  ...
49995  2016-03-27 14:38:19  2016-03-27 00:00:00  2016-04-01 13:47:40
49996  2016-03-28 10:50:25  2016-03-28 00:00:00  2016-04-02 14:18:02
49997  2016-04-02 14:44:48  2016-04-02 00:00:00  2016-04-04 11:47:27
49998  2016-03-08 19:25:42  2016-03-08 00:00:00  2016-04-05 16:45:07
49999  2016-03-14 00:42:12  2016-03-13 00:00:00  2016-04-06 21:18:48

[38629 rows x 3 columns]

In [21]:
autos["date_crawled"].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05 14:06:30    0.000026
2016-03-05 14:06:40    0.000026
2016-03-05 14:07:21    0.000026
2016-03-05 14:07:26    0.000026
2016-03-05 14:07:40    0.000026
                         ...   
2016-04-07 14:30:09    0.000026
2016-04-07 14:30:26    0.000026
2016-04-07 14:36:44    0.000026
2016-04-07 14:36:55    0.000026
2016-04-07 14:36:56    0.000026
Name: date_crawled, Length: 37561, dtype: float64

In [22]:
autos["ad_created"].value_counts(normalize=True, dropna=False).sort_index()

2015-06-11 00:00:00    0.000026
2015-08-10 00:00:00    0.000026
2015-09-09 00:00:00    0.000026
2015-11-10 00:00:00    0.000026
2015-12-30 00:00:00    0.000026
                         ...   
2016-04-03 00:00:00    0.039452
2016-04-04 00:00:00    0.037278
2016-04-05 00:00:00    0.011986
2016-04-06 00:00:00    0.003365
2016-04-07 00:00:00    0.001320
Name: ad_created, Length: 74, dtype: float64

In [23]:
autos["last_seen"].value_counts(normalize=True, dropna=False).sort_index().tail(12000)

2016-04-05 11:50:29    0.000026
2016-04-05 11:50:30    0.000026
2016-04-05 11:50:36    0.000026
2016-04-05 11:50:39    0.000026
2016-04-05 11:50:40    0.000026
                         ...   
2016-04-07 14:58:44    0.000078
2016-04-07 14:58:45    0.000026
2016-04-07 14:58:46    0.000026
2016-04-07 14:58:48    0.000078
2016-04-07 14:58:50    0.000078
Name: last_seen, Length: 12000, dtype: float64

In [24]:
l = (autos["last_seen"].value_counts(normalize=True, dropna=False).sort_index().tail(12000))

In [25]:
l.sum()

0.48932149421419147

The frequency of last three days being seen is around 50%, so there might be a selling surge in the last three days.

In [26]:
autos["registration_year"].describe()

count    38629.000000
mean      2005.678713
std         86.681928
min       1000.000000
25%       2001.000000
50%       2005.000000
75%       2009.000000
max       9999.000000
Name: registration_year, dtype: float64

In [27]:
autos["registration_year"].value_counts(dropna=False).sort_index(ascending=True)

1000    1
1001    1
1927    1
1929    1
1931    1
       ..
5911    1
6200    1
8888    1
9000    1
9999    2
Name: registration_year, Length: 91, dtype: int64

It is inlogical to have registration year as high as 9999 or as low as 1000, so we are going to set it from 1900 to 2021.

In [28]:
autos = autos[autos["registration_year"].between(1900,2021)]
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offertype  price   abtest vehicleType  registration_year    gearbox  \
0  privat   Angebot   5000  control         bus               2004    manuell   
1  privat   Angebot   8500  control   limousine               1997  automatik   
2  privat   Angebot   8990     test   limousine               2009    manuell   
3  privat   Angebot   4350  control  kleinwagen               2007  automatik   
4  privat   Angebot   1350     test       kombi               2003    manuell   

   powerPS   model  odometer_km  registration_month fuelType       brand  \
0      158  andere       150000                   3      lpg     peugeot   
1      286     7er       150000                   6   benzin         bmw   
2      102    golf        70000                   7   benzin  volkswagen   
3       71  fortwo        70000                   6   benzin       smart   
4        0   focus       150000                   7   benzin        ford   

  unrepaired_damage           ad_created  nrofpictures  postalCode  \
0              nein  2016-03-26 00:00:00             0       79588   
1              nein  2016-04-04 00:00:00             0       71034   
2              nein  2016-03-26 00:00:00             0       35394   
3              nein  2016-03-12 00:00:00             0       33729   
4              nein  2016-04-01 00:00:00             0       39218   

             last_seen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37  
3  2016-03-15 03:16:28  
4  2016-04-01 14:38:50

In [29]:
autos["registration_year"].value_counts(normalize = True).sort_index().head(50)

1927    0.000026
1929    0.000026
1931    0.000026
1934    0.000052
1937    0.000104
1938    0.000026
1939    0.000026
1941    0.000052
1943    0.000026
1948    0.000026
1950    0.000026
1951    0.000052
1952    0.000026
1953    0.000026
1954    0.000052
1955    0.000052
1956    0.000104
1957    0.000052
1958    0.000078
1959    0.000155
1960    0.000440
1961    0.000155
1962    0.000104
1963    0.000207
1964    0.000233
1965    0.000440
1966    0.000544
1967    0.000647
1968    0.000673
1969    0.000466
1970    0.000829
1971    0.000622
1972    0.000777
1973    0.000596
1974    0.000596
1975    0.000466
1976    0.000544
1977    0.000518
1978    0.001062
1979    0.000803
1980    0.001839
1981    0.000699
1982    0.000984
1983    0.001295
1984    0.001217
1985    0.002072
1986    0.001450
1987    0.001321
1988    0.002745
1989    0.003185
Name: registration_year, dtype: float64

In [30]:
autos["brand"].describe()

count          38616
unique            40
top       volkswagen
freq            8218
Name: brand, dtype: object

In [31]:
autos["brand"].value_counts(normalize= True, ascending= False)

volkswagen        0.212813
bmw               0.123783
mercedes_benz     0.110731
audi              0.096825
opel              0.089807
ford              0.058732
renault           0.037653
peugeot           0.027553
fiat              0.021442
skoda             0.018671
seat              0.018050
smart             0.016781
toyota            0.014528
mazda             0.014398
citroen           0.014036
nissan            0.013647
mini              0.010721
hyundai           0.010721
sonstige_autos    0.010229
volvo             0.008856
kia               0.007691
honda             0.007484
porsche           0.007225
mitsubishi        0.006862
chevrolet         0.006603
alfa_romeo        0.006215
suzuki            0.005697
dacia             0.003315
chrysler          0.003185
jeep              0.002693
land_rover        0.002564
jaguar            0.001813
daihatsu          0.001709
subaru            0.001683
saab              0.001372
daewoo            0.000958
trabant           0.000880
r

In [32]:
auto_price = autos["brand"].value_counts(normalize= True)
brand_select = auto_price[auto_price > 0.025].index

price_dict= {}

for a in brand_select:
    brand_rows = autos[autos["brand"] == a]
    price_mean = brand_rows["price"].mean()
    price_dict[a] = int(price_mean)

price_dict

{'volkswagen': 6798,
 'bmw': 9055,
 'mercedes_benz': 9215,
 'audi': 10200,
 'opel': 4177,
 'ford': 5731,
 'renault': 3538,
 'peugeot': 3921}

In [33]:
mileage_dict = {}

for i in brand_select:
    brand_rows = autos[autos["brand"] == a]
    mileage_mean = brand_rows["odometer_km"].mean()
    mileage_dict[i] = int(mileage_mean)
    
mileage_dict

{'volkswagen': 122542,
 'bmw': 122542,
 'mercedes_benz': 122542,
 'audi': 122542,
 'opel': 122542,
 'ford': 122542,
 'renault': 122542,
 'peugeot': 122542}

In [34]:
df = pd.DataFrame (pd.Series(price_dict), columns = ['mean_price'])
df

mean_price
volkswagen           6798
bmw                  9055
mercedes_benz        9215
audi                10200
opel                 4177
ford                 5731
renault              3538
peugeot              3921

In [35]:
mileage_series = pd.Series(mileage_dict)
df["mileage_mean"] = mileage_series
df

mean_price  mileage_mean
volkswagen           6798        122542
bmw                  9055        122542
mercedes_benz        9215        122542
audi                10200        122542
opel                 4177        122542
ford                 5731        122542
renault              3538        122542
peugeot              3921        122542